hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/HoodieRealtimeRecordReader.java

```diff
@@ -69,8 +69,18 @@ private static RecordReader<NullWritable, ArrayWritable> constructRecordReader(R
       LOG.info("Enabling merged reading of realtime records for split " + split);
       return new RealtimeCompactedRecordReader(split, jobConf, realReader);
     } catch (IOException ex) {
-      LOG.error("Got exception when constructing record reader", ex);
+      LOG.error("Got IOException when constructing record reader", ex);
       throw new HoodieException(ex);
+    } catch (Exception e) {
+      LOG.error("Got exception when constructing record reader", e);
+      try {
+        if (null != realReader) {
```

Comment:
Can we move the file handle closing logic into the finally block?

Reply:
You are right, the code is used to construct the reader.

Reply:
Sorry, I find my previous code implementation is not rigorous and do some changes. Please review again. @danny0405

Reply:
We generalize the catching of exception from `IOException` to `Exception`, is there any possibility other kinds of exception can throw here?

Reply:
![wecom-temp-b58c44dd244a2a664b31121bdfe28a60](https://github.com/apache/hudi/assets/48636225/d3051da0-92e1-4999-922f-b1fc7d87e4ba)
In this case, exception can not be catched. @danny0405